### PR TITLE
Monica/bug fixes

### PR DESCRIPTION
**Pull request by myo3**

### CODE REVIEW

File: TCAT/Cells/RouteTableViewCell.swift [hunk 1]
```diff
@@ -145,110 +184,131 @@ class RouteTableViewCell: UITableViewCell {
...
-                setDepartureTime(withDepartureTime: route.departureTime, withWalkingRoute: route.isRawWalkingRoute(), withSearchTime: searchTime, withSearchTimeType: searchTimeType)
+                self.setDepartureTimeAndLiveElements(withRoute: route, withSearchTime: self.searchTime, withSearchTimeType: self.searchTimeType)
             }
         }
     }
 
-
+    private func hideLiveElements() {
+        liveIndicatorView.setColor(to: .clear)
+        liveLabel.textColor = .clear
```
Comment (mattbarker016) [inline TCAT/Cells/RouteTableViewCell.swift:264]: better than white, nice

File: TCAT/Cells/RouteTableViewCell.swift [hunk 2]
```diff
@@ -145,110 +184,131 @@ class RouteTableViewCell: UITableViewCell {
...
+        liveLabel.text = "Late - \(Time.timeString(from: departureTime))"
+        liveIndicatorView.setColor(to: .liveRedColor)
+        
+        liveLabel.sizeToFit()
+        positionLiveIndicatorView(usingLiveLabel: liveLabel)
+    }
+    
+    private func setDepartureTime(withStartTime startTime: Date, withDepartureTime departureTime: Date, late: Bool) {
+        let boardTime = Time.timeString(from: startTime, to: departureTime)
+        departureTimeLabel.text = boardTime == "0 min" ? "Board now" : "Board in \(boardTime)"
```
Comment (mattbarker016) [inline TCAT/Cells/RouteTableViewCell.swift:292]: Should we discuss maybe some better language for this, since it's confusing?

Comment (myo3) [reply TCAT/Cells/RouteTableViewCell.swift:292]: yes wanted to talk about board in with yall

File: TCAT/Controllers/CustomNavigationController.swift [hunk 3]
```diff
@@ -63,18 +63,29 @@ class CustomNavigationController: UINavigationController, UINavigationController
...
+            NSAttributedStringKey.baselineOffset : 0.3
+        ]
+        let attributedString = NSMutableAttributedString(string: "  Back", attributes: attributes)
         backButton.setAttributedTitle(attributedString, for: .normal)
+        
         backButton.sizeToFit()
+        
+        // Expand frame to create bigger touch area
+        let leftInset: CGFloat = 30
+        backButton.frame = CGRect(x: backButton.frame.minX, y: backButton.frame.minY, width: backButton.frame.width + leftInset, height: backButton.frame.height + 100)
```
Comment (mattbarker016) [inline TCAT/Controllers/CustomNavigationController.swift]: abstract out all the constants for easier fixing if we want to tweak it

File: TCAT/Controllers/CustomNavigationController.swift [hunk 4]
```diff
@@ -63,18 +63,29 @@ class CustomNavigationController: UINavigationController, UINavigationController
...
-        // raise back button text a hair - attention to detail, baby
-        attributedString.addAttribute(NSAttributedStringKey.baselineOffset, value: 0.3, range: NSMakeRange(0, attributedString.length))
+        let backButton = UIButton()
+        backButton.setImage(#imageLiteral(resourceName: "back"), for: .normal)
+        backButton.tintColor = .primaryTextColor
         
+        let attributes: [NSAttributedStringKey : Any] = [
+            NSAttributedStringKey.font : UIFont(name: Constants.Fonts.SanFrancisco.Regular, size: 14.0)!,
+            NSAttributedStringKey.foregroundColor : UIColor.primaryTextColor,
+            NSAttributedStringKey.baselineOffset : 0.3
```
Comment (mattbarker016) [inline TCAT/Controllers/CustomNavigationController.swift:76]: nice refactor

File: TCAT/Views/DatePickerView.swift [hunk 5]
```diff
@@ -65,17 +66,22 @@ class DatePickerView: UIView {
 
     // MARK: Segement Control
 
+    @objc private func timeTypeSegmentedControlValueChanged(segmentControl: UISegmentedControl) {
+        let arriveBy = 1
```
Comment (mattbarker016) [inline TCAT/Views/DatePickerView.swift]: Do you have an enum for these? Maybe make the rawValue be int to help?

File: TCAT/Views/RouteSelectionView.swift [hunk 6]
```diff
@@ -145,9 +145,9 @@ class RouteSelectionView: UIView {
         var title = ""
         
         if Calendar.current.isDateInToday(date) || Calendar.current.isDateInTomorrow(date) {
-            let verb = (searchTimeType == .arriveBy) ? "Arrive" : "Leave" //Use simply,"arrive" or "leave"
-            let day = Calendar.current.isDateInToday(date) ? "" : " tomorrow" //if today don't put day
-            title = "\(verb)\(day) at \(Time.timeString(from: date))"
+            let verb = (searchTimeType == .arriveBy) ? "Arrive" : (searchTimeType == .leaveNow) ? "Leave now" : "Leave" //Use simply,"arrive" or "leave"
+            let day = Calendar.current.isDateInToday(date) ? "" : "tomorrow " //if today don't put day
+            title = (searchTimeType == .leaveNow) ? "\(verb) (\(day.capitalizingFirstLetter())\(Time.timeString(from: date)))" : "\(verb) \(day)at \(Time.timeString(from: date))"
```
Comment (mattbarker016) [inline TCAT/Views/RouteSelectionView.swift:150]: nice

**Review by mattbarker016 — COMMENTED** [review]
Cool. Let me know if you add anything else from our pod meeting later.

**Comment by mattbarker016** [comment]
Good to go

**Pull request merged**